control/sum.go

```diff
@@ -72,8 +96,11 @@ func (b *sum) reset() error {
 		}
 		b.operation[b.cfg.DependsOn[idx]] = sumOperand(c)
 	}
-	b.y = make([]*Signal, 1)
-	b.y[0] = makeSignal(b.cfg.Name)
+	b.y = make([]*Signal, len(b.cfg.DependsOn))
+	b.y[0] = makeSignal(b.cfg.DependsOn[0])
+	if len(b.operation) == 3 {
+		b.y[1] = makeSignal(b.cfg.DependsOn[1])
+	}
```

Comment:
hmm, why is the length three here? is it two feedback and one input?

Reply:
if I remember correctly its the two set points and then the endpoint. I will go back and re-figure out what's happening here and add some comments because I'm also kind of confused

Reply:
I don't know what I was doing here. changing to this:
```
b.y = make([]*Signal, len(b.cfg.DependsOn))
for i := range b.cfg.DependsOn {
	b.y[i] = makeSignal(b.cfg.DependsOn[i])
}
```